src/content/util/mapBlogArticlesToSections.util.ts

```diff
@@ -0,0 +1,17 @@
+import { BlogArticleMeta } from '@prisma/client'
+
+export const mapBlogArticlesToSections = (
+  articles: BlogArticleMeta[],
+  limit?: number,
+) => {
+  return Object.fromEntries(
+    articles.reduce((acc, curr) => {
+      const currentSectionSlug = curr.section.fields.slug
+      return acc.set(currentSectionSlug, [
+        ...(acc.has(currentSectionSlug)
+          ? [...(acc.get(currentSectionSlug) || []), curr].slice(0, limit)
+          : [curr]),
+      ])
+    }, new Map<string, BlogArticleMeta[]>()),
+  )
+}
```

Comment:
Can we make this more readable?
Something like this?

```
export const mapBlogArticlesToSections = (
  articles: BlogArticleMeta[],
  limit?: number,
) => {
  const sectionMap = new Map<string, BlogArticleMeta[]>();
  
  // Group articles by their section
  for (const article of articles) {
    const sectionSlug = article.section.fields.slug;
    
    // Get existing articles for this section or create empty array
    const existingArticles = sectionMap.get(sectionSlug) || [];
    
    const updatedArticles = [...existingArticles, article];
    
    const limitedArticles = limit ? updatedArticles.slice(0, limit) : updatedArticles;
    
    sectionMap.set(sectionSlug, limitedArticles);
  }
  
  return Object.fromEntries(sectionMap);
}

```

Reply:
I don't think introducing and additional scope and more variables with a `for-loop` approach is warranted here, so I kept the `.reduce` approach.

I was however able to break up the return value and was actually able to reduce the cyclical complexity since there's no need to check for `limit` since `.slice(0, undefined)` will give the entire array anyway, and spreading `[...[]]` results in `[]` anyway, so no need to check for existing array, just append to the end, even if it's empty ("point-free" programming FTW!)

Reply:
https://github.com/thegoodparty/gp-api/pull/186/commits/5d33ef3aba928b7910505a0bd85c91c9074e1ca7